sdk/client-core/core/src/main/java/io/clientcore/core/http/models/RequestOptions.java

```diff
@@ -131,6 +137,10 @@ public RequestOptions() {
      * @return The request callback.
      */
     public Consumer<HttpRequest> getRequestCallback() {
+        if (locked) {
+            throw LOGGER.logThrowableAsError(new IllegalStateException("RequestOptions.NONE is immutable."));
+        }
```

Comment:
It's kind of weird to throw an immutable exception when someone is calling as getter. In cases where the getter returns something immutable (like this `Consumer`) it is fine to just return it back. In cases where it returns something mutable (like `Context`), I suspect we should go down the rabbit-hole of making them lockable too - which you can talk to @alzimmermsft about.

Reply:
I think `Context` is immutable because every time you add something to it you get a new reference, so maybe we don't need these throws in any getter. I'll look into it.